controllers/agents/app/servicebinding_controller.go

```diff
@@ -170,6 +173,11 @@ func (r *ServiceBindingReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, err
 	}
 
+	if err = r.setOwnerReference(ctx, &serviceBinding, req.Namespace); err != nil {
+		reconcileLog.Error(err, "Failed to set owner reference on Servicebinding", "namespace", req.Namespace, "name", req.Name)
+		return ctrl.Result{}, err
+	}
+
```

Comment:
this should have priority over the logic above, please move it at L120

Reply:
done